sorana/src/main/java/org/java/mentorship/service/TestService.java

```diff
@@ -0,0 +1,74 @@
+package org.java.mentorship.service;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.java.mentorship.domain.Song;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Service
+@AllArgsConstructor
+public class TestService {
+
+    private List<Song> songs;
+
+    public List<Song> getSongs(){
+        return this.songs;
+    }
+
+    public List<Song> getSongsByStyle(String style){
+
+        if(style == null) return this.songs;
+
+        List<Song> songsByStyle = new ArrayList<>();
+
+        songsByStyle.stream()
+                .filter(song -> song.getStyle().equals(style))
+                .collect(Collectors.toList());
+        return songsByStyle;
+    }
+
+    public Song getSongById(int id){
+        if(id == 0) return null;
+        return this.songs.stream()
+                .filter(it-> it.getId() == id)
+                .findFirst().get();
+    }
+
+    public boolean addSong1(Song song){
+
+        int songId = 0;
+        if(!this.songs.isEmpty())
+            songId = this.songs.getLast().getId() + 1;
+
+        if(song.getArtistId() != 0 && song.getStyle() != null && song.getAlbumId()!=0){
+            return this.songs.add(new Song(
+                    songId,
+                    song.getStyle(),
+                    song.getArtistId(),
+                    song.getAlbumId()
+            ));
+        }
+
+
+        return false;
+    }
+
+    public List<Song> deleteSongById(int id){
+        this.songs.remove(this.getSongById(id));
+        return this.songs;
+    }
+
+    public Song updateSongById(int id, Song song){
+        this.songs.get(id).setStyle(song.getStyle());
```

Comment:
Not really sustainable if we had more fields

Maybe we could've stored the songs in a map and update the whole object in one go.
Additionally, you can add a validation that the ID actually exists, otherwise we'll get a NPE

Nonetheless, if it works it's good